test/test_suite.py

```diff
@@ -837,3 +835,91 @@ def test_nolength_binary(self):
 @pytest.mark.skip("Spanner doesn't support quotes in table names.")
 class QuotedNameArgumentTest(_QuotedNameArgumentTest):
     pass
+
+
+@pytest.mark.skip("Spanner doesn't support IS DISTINCT FROM clause")
+class IsOrIsNotDistinctFromTest(_IsOrIsNotDistinctFromTest):
+    pass
+
+
+class OrderByLabelTest(_OrderByLabelTest):
+    @pytest.mark.skip(
+        "Spanner throws a syntax error as group by composed doesn't support."
+    )
+    def test_group_by_composed(self):
+        pass
+
+
+class CompoundSelectTest(_CompoundSelectTest):
+    @pytest.mark.skip("Spanner throws a syntax error.")
```

Comment:
Please reword these to `Spanner doesn't support <unsupported statement>.`

Reply:
Running these locally, I am getting the syntax error however I think the intended SQL statement should be supported. It looks like the argument replacement is not working as intended.

Error:
```
sqlalchemy.exc.ProgrammingError: (google.cloud.spanner_dbapi.exceptions.ProgrammingError) 400 Syntax error: Expected end of input but got keyword WHERE [at 8:12]\n LIMIT @a4 WHERE 1=1\n
[SQL: (SELECT some_table.id, some_table.x, some_table.y 
FROM some_table 
WHERE some_table.id = %s ORDER BY some_table.id
 LIMIT %s) UNION DISTINCT (SELECT some_table.id, some_table.x, some_table.y 
FROM some_table 
WHERE some_table.id = %s ORDER BY some_table.id
 LIMIT %s) ORDER BY id
 LIMIT %s]
[parameters: [2, 1, 3, 1, 2]]
```
Valid SQL query in Spanner:
```
(SELECT some_table.id, some_table.x, some_table.y 
FROM some_table 
WHERE some_table.id = 2 ORDER BY some_table.id
 LIMIT 1) UNION DISTINCT (SELECT some_table.id, some_table.x, some_table.y 
FROM some_table 
WHERE some_table.id = 3 ORDER BY some_table.id
 LIMIT 1) ORDER BY id
 LIMIT 2
```

This may be a bug in DBAPI. Could you look into this?

Reply:
sure.

Reply:
@larkee Query is not valid in spanner which you have mentioned above, it throws an error due to `GROUP BY` and `LIMIT` in subqueries and  brackets `()`  used for second subqueries.

Reply:
The query with the placeholders replaced by the parameters is valid, as per Pantheon:
![image](https://user-images.githubusercontent.com/31196561/118499044-ed8fb780-b769-11eb-8f87-751b089f315e.png)

However `spanner_dbapi` is throwing an error for the query, which indicates there is a bug in the `spanner_dbapi` logic.

From the exception:
```
400 Syntax error: Expected end of input but got keyword WHERE [at 8:12]\n LIMIT @a4 WHERE 1=1\n
```
It looks like a dummy `WHERE` clause is being added incorrectly.

The statement is being classified as `STMT_UPDATING` rather than `STMT_NON_UPDATING` as the regex expressions currently do not account for SQL that begins with brackets.

Reply:
I've opened an issue to track this bug so if you update the skip reason to reference the issue then I'm happy to approve and merge this PR.

The issue: https://github.com/googleapis/python-spanner/issues/347